packages/common/src/hooks/usePremiumContent.ts

```diff
@@ -88,11 +98,15 @@ export const usePremiumContentAccessMap = (tracks: Partial<Track>[]) => {
 export const usePremiumConditionsEntity = (
   premiumConditions: Nullable<PremiumConditions>
 ) => {
-  const {
-    follow_user_id: followUserId,
-    tip_user_id: tipUserId,
-    nft_collection: nftCollection
-  } = premiumConditions ?? {}
+  const followUserId = isPremiumContentFollowGated(premiumConditions)
+    ? premiumConditions?.follow_user_id
+    : null
+  const tipUserId = isPremiumContentTipGated(premiumConditions)
+    ? premiumConditions?.tip_user_id
+    : null
+  const nftCollection = isPremiumContentCollectibleGated(premiumConditions)
+    ? premiumConditions?.nft_collection
+    : null
 
```

Comment:
ternaries here

Reply:
still getting type errors even if I null check first. gonna stick with the type guard for now but let's chat about it tmr.

Reply:
I can live with the type guards!